### PR TITLE
Migrate Line/LineString to be a series of Coordinates (not Points).

### PR DESCRIPTION
**Pull request by frewsxcv**
Relevant conversation in https://github.com/georust/rust-geo/issues/15.

This is a breaking change for `geo-types`.

I'm _not_ planning to publish a release if this merges. I have a couple other breaking changes I'd like to consider before bumping the Rust geo ecosystem.

### CODE REVIEW

**Comment by frewsxcv** [comment]
A pretty big pull request, but it should be ready to go now! I'll squash before approving

File: geo-types/src/coordinate.rs [hunk 1]
```diff
@@ -12,6 +12,50 @@ where
     pub y: T,
 }
 
+impl<T> Coordinate<T>
+where
+    T: CoordinateType,
+{
+    /// Returns the dot product of the two points:
```
Comment (urschrei) [inline geo-types/src/coordinate.rs]: should this be "…of the two coordinates"?

File: geo-types/src/coordinate.rs [hunk 2]
```diff
@@ -12,6 +12,50 @@ where
...
+    /// let coord = Coordinate { x: 1.5, y: 0.5 };
+    /// let dot = coord.dot(Coordinate { x: 2.0, y: 4.5 });
+    ///
+    /// assert_eq!(dot, 5.25);
+    /// ```
+    pub fn dot(&self, coord: Coordinate<T>) -> T {
+        self.x * coord.x + self.y * coord.y
+    }
+
+    /// Returns the cross product of 3 points. A positive value implies
```
Comment (urschrei) [inline geo-types/src/coordinate.rs]: Same here.

File: geo-types/src/line.rs [hunk 3]
```diff
@@ -141,8 +141,27 @@ where
     /// # );
     /// ```
     ///
+    /// [determinant]: https://en.wikipedia.org/wiki/Determinant
```
Comment (urschrei) [inline geo-types/src/line.rs]: FYI: links have to be inline for RLS

File: geo-types/src/line_string.rs [hunk 4]
```diff
@@ -1,111 +1,150 @@
...
+/// let mut coords_iter = vec![
+///     Coordinate { x: 0., y: 0. },
+///     Coordinate { x: 10., y: 0. }
+/// ].into_iter();
+///
+/// let line_string: LineString<f32> = coords_iter.collect();
 /// ```
 ///
-/// You can iterate over the points in the `LineString`
+/// You can iterate over the coordintes in the `LineString`
```
Comment (urschrei) [inline geo-types/src/line_string.rs]: Typo.

**Comment by frewsxcv** [comment]
bors r=urschrei

**Pull request merged**